htdocs/comm/propal/class/propal.class.php

```diff
@@ -2060,7 +2060,9 @@ public function set_date($user, $date, $notrigger = 0)
 
 			$this->db->begin();
 
-			$sql = "UPDATE ".MAIN_DB_PREFIX."propal SET datep = '".$this->db->idate($date)."'";
+			$fin_validite = $date + ($this->duree_validite * 24 * 3600);
```

Comment:
If $this->duree_validite is not set and in database fin_validite is empty, we must keep field empty (unknown) in database.